Failed to build with Visual Studio 2019 Community

### Describe the bug

I was trying to follow [GettingStarted]( https://github.com/microsoft/ebpf-for-windows/blob/master/docs/GettingStarted.md) to build the project with Visual Studio 2109 Community.  Build failed.



My Dev Environment:
- VM on Azure
```
OS Name:                   Microsoft Windows 10 Pro
OS Version:                10.0.19044 N/A Build 19044
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
```

- Visual Studio 
```
Microsoft Visual Studio Community 2019
Version 16.11.15
VisualStudio.16.Release/16.11.15+32510.428
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Community

Visual C++ 2019   00435-00000-00000-AA081
Microsoft Visual C++ 2019

ASP.NET and Web Tools 2019   16.11.106.23128
ASP.NET and Web Tools 2019

C# Tools   3.11.0-4.22108.8+d9bef045c4362fbcab27ef35daec4e95c8ff47e1
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

IntelliCode Extension   1.0
IntelliCode Visual Studio Extension Detailed Info

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

NuGet Package Manager   5.11.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Test Adapter for Boost.Test   1.0
Enables Visual Studio's testing tools with unit tests written for Boost.Test.  The use terms and Third Party Notices are available in the extension installation directory.

Test Adapter for Google Test   1.0
Enables Visual Studio's testing tools with unit tests written for Google Test.  The use terms and Third Party Notices are available in the extension installation directory.

TypeScript Tools   16.0.30526.2002
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   3.11.0-4.22108.8+d9bef045c4362fbcab27ef35daec4e95c8ff47e1
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Tools for CMake   1.0
Visual Studio Tools for CMake
```

### OS information

_No response_

### Steps taken to reproduce bug

Following [GettingStarted]( https://github.com/microsoft/ebpf-for-windows/blob/master/docs/GettingStarted.md) to build the project with Visual Studio 2109 Community. 

Build failed with the step `nuget.exe restore ebpf-for-windows.sln` 

### Expected behavior

Build Successfully 

### Actual outcome

Build failed with the step `nuget.exe restore ebpf-for-windows.sln` 
```
C:\Users\song\ebpf-for-windows>c:\pkgs\downloads\nuget.exe restore ebpf-for-windows.sln
MSBuild auto-detection: using msbuild version '16.11.2.50704' from 'C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\MSBuild\Current\Bin'.
C:\Users\song\ebpf-for-windows\external\ebpf-verifier\build\ebpfverifier.vcxproj(32,3): error MSB4019: The imported project "C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\Microsoft.Cpp.Default.props" was not found. Confirm that the expression in the Import declaration "C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\\Microsoft.Cpp.Default.props" is correct, and that the file exists on disk.
```

### Additional details

Hit the same issue with Visual Studio 2019 professional as well.

That's very odd. I am still seeing errors in the log related to missing files from boost, which should have been populated when the  nuget package was restored. 

The first fatal error is:
```
C:\1\ebpf-for-windows\external\ebpf-verifier\src\crab_utils/bignums.hpp(10,10): fatal error C1083: Cannot open include file: 'boost/functional/hash.hpp': No such file or directory
```

Can you check if the following file is present (from the root of your repo):
packages\boost.1.75.0.0\lib\native\include\boost\functional\hash.hpp

@Alan-Jowett Thanks for your help!
Following the steps you outlined above, I managed to build the latest code. 
```
    297 Warning(s)
    0 Error(s)

Time Elapsed 00:06:05.39
```

I had to work around two issues. 
- Failed to start VS installer after `alternative install steps` with the error message `Another instance is running....`. I restarted the VM and the problem went away. 

- Seen error message while building the binary. This issue has been resolved by reinstalling WDK.vsix. 
```
     C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppBuild.tar 

       gets(439,5): error MSB8020: The build tools for WindowsKernelModeDriver10.0 (Platform Toolset = 'WindowsKernelMo 

       deDriver10.0') cannot be found. To build using the WindowsKernelModeDriver10.0 build tools, please install Windo 

       wsKernelModeDriver10.0 build tools.  Alternatively, you may upgrade to the current Visual Studio tools by select 

       ing the Project menu or right-click the solution, and then selecting "Retarget solution". [c:\1\ebpf-for-windows 

       \libs\ubpf\kernel\ubpf_kernel.vcxproj] 
```

--- a/docs/GettingStarted.md
+++ b/docs/GettingStarted.md
@@ -15,7 +15,7 @@ The following must be installed in order to build this project:
    which must be selected as an Individual component in the VS installer
 3. [Visual Studio Build Tools 2019](https://aka.ms/vs/16/release/vs_buildtools.exe)
 4. [WDK for Windows 10, version 2004](https://go.microsoft.com/fwlink/?linkid=2128854), including the "Windows Driver Kit Visual Studio extension" (make sure the check box is checked before completing the installer)
-5. [Clang for Windows 64-bit version 11.0.1](https://github.com/llvm/llvm-project/releases/download/llvmorg-11.0.1/LLVM-11.0.1-win64.exe) or [The latest release of Clang for Windows 64-bit](https://github.com/llvm/llvm-project/releases/latest)
+5. [Clang for Windows 64-bit version 11.0.1](https://github.com/llvm/llvm-project/releases/download/llvmorg-11.0.1/LLVM-11.0.1-win64.exe).  Note: clang versions 12 and higher are NOT yet supported, as they perform program optimizations that are incompatible with the PREVAIL verifier.
 6. [nuget.exe](https://www.nuget.org/downloads) installed to a location such as "C:\Program Files (x86)\NuGet\"
 
 You should add the paths to `git.exe`, `cmake.exe` and `nuget.exe` to the Windows PATH environment variable after the software packages above have been installed.

